### ISSUE TITLE
[Bug]: setDelaySelectAllEnabled works on "Select all" link but doesn't on the select all checkbox

### ISSUE THREAD

**Issue by matberry** [rappasoft/laravel-livewire-tables#1996]
### What happened?

I used `$this->setDelaySelectAllEnabled();` in `configure()`. 
When i click on "Select all" link, `$this->setDelaySelectAllEnabled();` behavior is ok, but when i select all via the checkbox 
![image](https://github.com/user-attachments/assets/e942001e-87e6-491d-8f4d-f234d8cfbea7)
it fires to the backend immediatly.

### How to reproduce the bug

- Have a large amount of rows
- `$this->setDelaySelectAllEnabled();` in `configure()`
- ✅ select one row, then click on "Select all" link
- ❌ click on the select all checkbox

### Package Version

3.4.21

### PHP Version

8.3.x

### Laravel Version

11.25.0

### Alpine Version

3.14.1

### Theme

Tailwind 3.x

**Comment by matberry** [comment]
The issue is still a problem.

**Comment by lrljoe** [comment]
Yep, still on the radar, just likely won't be an imminent fix from my perspective. Happy to take a PR tho
